### PR TITLE
Replace linearTimeSeries with cartesianChart

### LINKED ISSUE #252: Replace linearTimeSeries with cartesianChart

### PR DESCRIPTION
**Pull request by djmiley**
Resolves #252

### CODE REVIEW

File: src/assets/js/chart/nav.js [hunk 1]
```diff
@@ -4,23 +4,28 @@
...
+            });
 
         var viewScale = fc.scale.dateTime();
 
         var area = fc.series.area()
-            .yValue(function(d) { return d.open; });
+            .yValue(function(d) { return d.close; });
         var line = fc.series.line()
-            .yValue(function(d) { return d.open; });
+            .yValue(function(d) { return d.close; });
```
Comment (djmiley) [inline src/assets/js/chart/nav.js]: Changed these for consistency with primaryChart

**Comment by djmiley** [comment]
xXscale() is https://github.com/ScottLogic/d3fc/issues/564

**Comment by jleft** [comment]
Looking good - I think the styling on the latest price annotation text has been lost though?

File: src/assets/js/chart/primary.js [hunk 2]
```diff
@@ -94,7 +94,7 @@
             .mapping(function(series) {
                 switch (series) {
                     case closeLine:
-                        return [this.data[this.data.length - 1]];
+                        return this.data[this.data.length - 1];
```
Comment (jleft) [inline src/assets/js/chart/primary.js]: I'm not sure why this has changed - I think we'd still want to map an array?

File: src/assets/js/chart/primary.js [hunk 3]
```diff
@@ -148,18 +156,19 @@
...
             // Find current tick values and add close price to this list, then set it explicitly below
             var latestPrice = currentYValueAccessor(model.data[model.data.length - 1]);
-            var tickValues = produceAnnotatedTickValues(timeSeries.yScale(), [latestPrice]);
-            timeSeries.yTickValues(tickValues)
+            var tickValues = produceAnnotatedTickValues(yScale, [latestPrice]);
+            primaryChart.yTickValues(tickValues)
                 .yDecorate(function(s) {
-                    s.filter(function(d) { return d === latestPrice; })
+                    var yAxisTicks = s.selectAll('.tick');
+                    yAxisTicks.filter(function(d, i) { return d === latestPrice; })
```
Comment (jleft) [inline src/assets/js/chart/primary.js]: Can chain this method with the previous one (also don't need `i`):

```js
s.selectAll('.tick')
    .filter(function(d) { return d === latestPrice; })
    // etc.
```

File: src/assets/js/chart/primary.js [hunk 4]
```diff
@@ -148,18 +156,18 @@
...
 
             // Find current tick values and add close price to this list, then set it explicitly below
             var latestPrice = currentYValueAccessor(model.data[model.data.length - 1]);
-            var tickValues = produceAnnotatedTickValues(timeSeries.yScale(), [latestPrice]);
-            timeSeries.yTickValues(tickValues)
+            var tickValues = produceAnnotatedTickValues(yScale, [latestPrice]);
+            primaryChart.yTickValues(tickValues)
                 .yDecorate(function(s) {
-                    s.filter(function(d) { return d === latestPrice; })
+                    s.selectAll('.tick').filter(function(d) { return d === latestPrice; })
```
Comment (jleft) [inline src/assets/js/chart/primary.js]: `.filter` on a new line, please, for consistency.

**Comment by jleft** [comment]
LGTM :+1: thanks @djmiley

**Pull request merged**